projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3021,8 +3023,12 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
                 return;
             }
 
-            if (this.overlayService.getOverlayById(event.id)?.settings?.outlet === this.outlet &&
-                this.overlayIDs.indexOf(event.id) < 0) {
+            // do not hide the overlay if it's attached to a row
+            if (this.rowEditingOverlay.overlayId === event.id) {
```

Comment:
Should also check if this.rowEditingOverlay exists since it probably won't in grids that don't have row editing.